python/ml4ir/base/model/scoring/monte_carlo_scorer.py

```diff
@@ -0,0 +1,123 @@
+        interaction_model : `InteractionModel` object
+            InteractionModel that defines the feature transformation layers
+            on the input model features
+        loss : `RelevanceLossBase` object
+            Relevance loss object that defines the final activation layer
+            and the loss function for the model
+        file_io : `FileIO` object
+            FileIO object that handles read and write
+        aux_loss : `RelevanceLossBase` object
+            Auxiliary loss to be used in conjunction with the primary loss
+        aux_loss_weight: float
+            Floating point number in [0, 1] to indicate the proportion of the auxiliary loss
+            in the total final loss value computed using a linear combination
+            total loss = (1 - aux_loss_weight) * loss + aux_loss_weight * aux_loss
+        aux_metrics: List of keras.metrics.Metric
+            Keras metric list to be computed on the aux label
+        output_name : str, optional
+            Name of the output that captures the score computed by the model
+        logger : Logger, optional
+            Logging handler
+        logs_dir : str, optional
+            Path to the logging directory
+
+        Notes
+        -----
+        logs_dir : Used to point model architectures to local logging directory,
+            primarily for saving visualizations.
+        """
+        super().__init__(feature_config=feature_config,
+                         model_config=model_config,
+                         interaction_model=interaction_model,
+                         loss=loss,
+                         aux_loss=aux_loss,
+                         aux_loss_weight=aux_loss_weight,
+                         aux_metrics=aux_metrics,
+                         output_name=output_name,
+                         logger=logger,
+                         file_io=file_io,
+                         logs_dir=logs_dir,
+                         **kwargs)
+        self.monte_carlo_inference_trials = self.model_config[MonteCarloInferenceKey.MONTE_CARLO_INFERENCE_TRIALS][MonteCarloInferenceKey.NUM_TRIALS]
+        self.monte_carlo_inference_trials_tf = tf.constant(self.monte_carlo_inference_trials + 1, dtype=tf.float32)
+
+    def call(self, inputs: Dict[str, tf.Tensor], training=None):
+        """
+        Compute score from input features
+
+        Parameters
+        --------
+        inputs : dict of tensors
+            Dictionary of input feature tensors
+
+        Returns
+        -------
+        scores : dict of tensor object
+            Tensor object of the score computed by the model
+        """
+        # TODO replace loop with vector operations for performance
+        scores = super().call(inputs, training=False)[self.output_name]
+        for _ in range(self.monte_carlo_inference_trials):
+            scores += super().call(inputs, training=True)[self.output_name]
+        scores = tf.divide(scores, self.monte_carlo_inference_trials_tf)
```

Comment:
@mohazahran shouldn't you be dividing by num trials + 1?

Reply:
Ah I see you've already added it above. Perhaps cleaner to create constant in constructor and do the addition here to avoid confusion?

Reply:
In the constructor I have two variables, one if the number of MC trials and the other is the one used in the normalization in the call. I added a comment to explain that